sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -925,19 +925,19 @@ private[hive] class HiveClientImpl(
   }
 
   override def createFunction(db: String, func: CatalogFunction): Unit = withHiveState {
-    shim.createFunction(client, db, func)
+    shim.createFunction(client, db, func, userName)
   }
 
   override def dropFunction(db: String, name: String): Unit = withHiveState {
     shim.dropFunction(client, db, name)
   }
 
   override def renameFunction(db: String, oldName: String, newName: String): Unit = withHiveState {
-    shim.renameFunction(client, db, oldName, newName)
+    shim.renameFunction(client, db, oldName, newName, userName)
   }
 
   override def alterFunction(db: String, func: CatalogFunction): Unit = withHiveState {
-    shim.alterFunction(client, db, func)
+    shim.alterFunction(client, db, func, userName)
```

Comment:
Do we change owner every time when someone alters it?

Reply:
I checked, Hive/Spark does not have the syntax of alter function, although the api is provided in the hive meta store.
https://cwiki.apache.org/confluence/display/Hive/LanguageManual+DDL#LanguageManualDDL-CreateFunction

If `alterFunction` does not pass the owner name, this will lose the owner name.

Or do not pass the current user name when `alterFunction`?

Reply:
we shall keep the original owner name

Reply:
`renameFunction` can keep the original owner name, and the function definition has been modified except for the function name of `alterFunction`.